spec/features/projects_manage_spec.rb

```diff
@@ -427,9 +432,39 @@
     end
   end
 
+  describe "when unlocking a project" do
+    context "when a user is an admin" do
+      let(:user) { FactoryBot.create(:user, admin: true) }
+
+      context "if a project is locked" do
+        let(:locked_project) { FactoryBot.create(:project, :locked) }
+
+        it "shows an unlocked button" do
+          visit project_path(locked_project)
+          within "div.btn-group.actions" do
+            expect(page).to have_text("Unlock Project")
+          end
+        end
+
+        it "unlocks a project" do
+          visit project_path(locked_project)
+
+          click_button "Unlock Project"
+
+          within "div.btn-group.actions" do
+            expect(page).to have_text("Lock Project")
+            expect(page).not_to have_text("Unlock Project")
+          end
+        end
+      end
+    end
+  end
+
   def expect_buttons_to_be_hidden
     ["Delete Project", "Lock Project", "Add Sub-Project", "Add a Story"].each do |btn|
-      expect(page).not_to have_selector(:link_or_button, btn)
+      within "div.btn-group.actions" do
```

Comment:
I think this is a risky change because I am not sure all buttons are within the btn-group div? Or if we add buttons in the future outside this div, then this test will not fail even if it should.

Reply:
Currently the buttons are within that div but I can see how that would be problematic. Would moving the scope up to the div.container be more optimal?